pcap.pyx

```diff
@@ -227,92 +237,85 @@ cdef class pcap:
     property filter:
         """Current packet capture filter."""
         def __get__(self):
-            return self.__filter
-    
+            return self.__filter.decode('UTF-8')
+
     property fd:
         """File descriptor (or Win32 HANDLE) for capture handle."""
         def __get__(self):
             return pcap_ex_fileno(self.__pcap)
-        
+
     def fileno(self):
         """Return file descriptor (or Win32 HANDLE) for capture handle."""
         return pcap_ex_fileno(self.__pcap)
-    
+
     def setfilter(self, value, optimize=1):
         """Set BPF-format packet capture filter."""
         cdef bpf_program fcode
         free(self.__filter)
-        self.__filter = strdup(value)
+        py_byte_value = value.encode('UTF-8')
+        self.__filter = strdup(py_byte_value)
```

Comment:
Yeah what I meant is that since you are anyways going from `unicode`->`bytes`->`unicode` when calling `setfilter()` and then accessing `filter`, maybe it makes sense to avoid the intermediate conversion and just use bytes all around.
I don't actually have a very strong opinion for this and I guess an argument in favour of using `unicode` is that these are things typed by users and displayed so they should actually have an encoding attached to them, but maybe the headaches that could arise from doing this do not justify this benefit.

I had this issue with a library I was using that added support for python3 and it broke in peculiar ways client code using it from py2 since I was expecting things to be `str` and they were suddenly all `unicode`s